tests/dummy_tests.rs

```diff
@@ -174,6 +174,31 @@ mod already_connected {
 
     use tokio_util::codec::{Framed, LinesCodec};
 
+    #[tokio::test]
+    async fn back_to_back_shutdown_attempts() {
```

Comment:
I like this test, it is good to have, but I do not think it belongs in this file. This is a real test that uses (local) network communications. I  think it makes sense to put it in a different file, maybe called `integration_tests.rs`

Reply:
Fixed in 1a35332